pkg/utils/maps/maps.go

```diff
@@ -197,3 +197,14 @@ func SerializeMap(m map[string]string) string {
 func DeSerializeCache(s string) []string {
 	return strings.Split(s, ",")
 }
+
+// GetIndexFromMap returns the key and the value of the map at the given index.
+func GetIndexFromMap[K comparable, V any](m map[K]V, i int) (k K, v V) {
+	for k, v = range m {
+		if i == 0 {
+			return k, v
+		}
+		i--
+	}
+	return *new(K), *new(V)
+}
```

Comment:
Looking at the official go documentation (https://go.dev/ref/spec#RangeClause) the order of iteration is not guaranteed to be the same every time (it is only when the map printed). Is this safe in our case?